source/documentation/appendix-docs/botor.md

```diff
@@ -319,37 +424,135 @@ write_df_to_csv_in_s3(
+  if(!any(grepl('data.frame', class(df)))) {
+    stop("df entered isn't a valid dataframe object")
+  }
+  if(tools::file_ext(s3_path) != 'csv') {
+    stop("s3_path entered is either not a csv or is missing the .csv suffix")
+  }
+  # trim s3:// if included by the user - removed so we can supply both 
+  # alpha-... and s3://alpha - and then add again
+  s3_path <- paste0("s3://", gsub('^s3://', "", s3_path))
+  if(!overwrite & botor::s3_exists(s3_path)) {
+    stop("s3_path entered already exists and overwrite is FALSE")
+  }
+  # write csv
+  botor::s3_write(df, fun = write.table, uri = s3_path, ...)
+}
+```
+
+##### Examples
 
+```r
+write_df_to_table_in_s3(
+    df = mtcars, 
+    s3_path = "alpha-everyone/mtcars_boto.csv"
+)
+write_df_to_table_in_s3(
+    df = mtcars, 
+    s3_path = "alpha-everyone/mtcars_boto.csv", 
+    row.names = FALSE
+)
 ```
-download_file_from_s3 <- function(s3_path, local_path, overwrite = FALSE) {
-  
-  # trim s3:// if included by the user
-  s3_path <- gsub('^s3://',"",s3_path)
-  
-  # add s3:// back in where required
-  s3_path <- paste0('s3://', s3_path)
-  
-  if (!(file.exists(local_path)) || overwrite) {
-    # download file
-    botor::s3_download_file(uri = s3_path, file = local_path, force = overwrite)
+
+#### `write_file_to_s3`
+
+```r
+write_file_to_s3 <- function(local_file_path, s3_path, overwrite=FALSE, 
+                             multipart = "unused") {
+  s3_path <- paste0("s3://", gsub("^s3://", "", s3_path))
+  if (overwrite || !(s3_file_exists(s3_path))) {
+    tryCatch(
+      botor::s3_upload_file(local_file_path, full_s3_path(s3_path)),
+      error = function(c) {
+        message(glue::glue("Could not upload {local_file_path} to {s3_path}"),
+                appendLF = TRUE)
+        stop(c, appendLF = TRUE)
+      }
+    )
   } else {
-    stop("The file already exists locally and you didn't specify overwrite=TRUE")
+    stop("File already exists and you haven't set overwrite = TRUE, stopping")
+  }
+}
+```
+
+##### Examples
+
+```r
+write_file_to_s3("my_local_data.csv", "s3://alpha-mybucket/my_data.csv")
+```
+
+#### `list_files_in_buckets`
+
+```r
+list_files_in_buckets <- function(bucket_filter = NULL, prefix = NULL,
+                                  path_only = FALSE, max = "unused") {
+  if (is.null(bucket_filter)) {
+    stop(paste0("You must provide one or more buckets e.g. ",
+                "accessible_files_df('alpha-everyone')  This function will ",
+                "list their contents"))
   }
-  
+  if(!is.character(bucket_filter)) {
+    stop("Supplied bucket_filter is not of class: character")
+  }
+  if(!is.character(prefix)&!is.null(prefix)) {
+    stop("Supplied prefix is not of class: character")
+  }
+  list_files_in_bucket <- function(bucket) {
+    # trim s3:// if included by the user - removed so we can supply both
+    # alpha-... and s3://alpha-...
+    bucket <- gsub('^s3://', "", bucket)
+    cols_to_keep <- c("key","last_modified","size","bucket_name")
```

Comment:
As I found trying to use something similar earlier - we need to:
1) Rename some of these columns to match the original s3tools version
2) Use some regex (or something else) to pull out our filenames

Reply:
I believe the columns we are currently keeping match the following:
`c("key", "lastmodified", "size", "bucket")`